Assets/Scripts/DoDamage.cs

```diff
@@ -5,13 +5,16 @@
 public class DoDamage : MonoBehaviour
 {
     public GameObject damageEffect;
-    private List<Collision> collisions; //Declare list with all colliders that are colliding with the weapon.
-    private Animator animator; // Declare the animator of the weapon
+    private List<Collision> _collisions; //Declare list with all colliders that are colliding with the weapon.
+    public Animator animator; // Declare the animator of the weapon
 
     void Start()
     {
-        collisions = new List<Collision>(); //Create new List
-        animator = gameObject.GetComponentInParent<Animator>();
+        _collisions = new List<Collision>(); //Create new List
+        if (gameObject.GetComponentInParent<Animator>() != null)
+            animator = gameObject.GetComponentInParent<Animator>();
+        else
+            animator = null;
```

Comment:
dit lijkt me niet nodig toch? als hij niet gevonden kan worden is hij altijd null